api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -206,6 +202,38 @@ message ExternalProcessor {
   // Instead, the stream to the external processor will be closed. There will be no
   // more external processing for this stream from now on.
   bool disable_immediate_response = 15;
+
+  // Options related to the sending and receiving of dynamic metadata
+  MetadataOptions metadata_options = 16;
+}
+
+// The MetadataOptions structure defines options for the sending and receiving of
+// dynamic metadata. Specifically, which namespaces to send to the server, whether
+// metadata returned by the server may be written, and how that metadata may be written.
+message MetadataOptions {
+  message MetadataNamespacesToForward {
+    // Specifies a list of metadata namespaces whose values, if present,
+    // will be passed to the ext_proc service as an opaque *protobuf::Struct*.
+    repeated string untyped = 1;
+
+    // Specifies a list of metadata namespaces whose values, if present,
+    // will be passed to the ext_proc service as a *protobuf::Any*. This allows
+    // envoy and the external processing server to share the protobuf message
+    // definition for safe parsing.
+    repeated string typed = 2;
+  }
+
+  // Describes which typed or untyped dynamic metadata namespaces to forward to
+  // the external processing server
+  MetadataNamespacesToForward forwarding_namespaces = 1;
+
+  // If set to true, metadata returned from the external processing server will
+  // be written into the stream's dynamic metadata.
+  bool enable_returned_metadata = 2;
```

Comment:
why cant this be namespaces_to_forward, namespaces_to_accept ?

Reply:
Is it desired to allow writing to arbitrary metadata namespaces?

Reply:
Is there a specific use-case you're thinking of? It seems to me that it is safer and more consistent to keep all of the returned metadata under the ext_proc namespace to prevent the external service from overwriting internal filters' data